cli/src/commands.rs

```diff
@@ -0,0 +1,377 @@
+	Listen {
+		/// exit after given number of Integritee events
+		#[clap(short, long = "exit-after")]
+		events: Option<u32>,
+
+		/// exit after given number of blocks
+		#[clap(short, long = "await-blocks")]
+		blocks: Option<u32>,
+	},
+
+	/// Transfer funds from an on-chain account to an incognito account
+	ShieldFunds {
+		/// Sender's on-chain AccountId in ss58check format
+		from: String,
+
+		/// Recipient's incognito AccountId in ss58check format
+		to: String,
+
+		/// Amount to be transferred
+		amount: Balance,
+
+		/// Shard identifier
+		shard: String,
+	},
+
+	/// trusted calls to worker enclave
+	#[clap(after_help = "stf subcommands depend on the stf crate this has been built against")]
+	Trusted(TrustedArgs),
+}
+
+pub fn match_command(cli: &Cli) {
+	match &cli.command {
+		Commands::Balance { account } => balance(cli, account),
+		Commands::NewAccount => new_account(),
+		Commands::ListAccounts => list_accounts(),
+		Commands::PrintMetadata => print_metadata(cli),
+		Commands::PrintSgxMetadata => print_sgx_metadata(cli),
+		Commands::Faucet { accounts } => faucet(cli, accounts),
+		Commands::Transfer { from, to, amount } => transfer(cli, from, to, amount),
+		Commands::ListWorkers => list_workers(cli),
+		Commands::Listen { events, blocks } => listen(cli, events, blocks),
+		Commands::ShieldFunds { from, to, amount, shard } =>
+			shield_funds(cli, from, to, amount, shard),
+		Commands::Trusted(trusted) => trusted_commands::match_trusted_commands(cli, trusted),
+	};
+}
+
+fn balance(cli: &Cli, account: &str) {
+	let api = get_chain_api(cli);
+	let accountid = get_accountid_from_str(account);
+	let balance =
+		if let Some(data) = api.get_account_data(&accountid).unwrap() { data.free } else { 0 };
+	println!("{}", balance);
+}
+
+fn new_account() {
+	let store = LocalKeystore::open(PathBuf::from(&KEYSTORE_PATH), None).unwrap();
+	let key: sr25519::AppPair = store.generate().unwrap();
+	drop(store);
+	println!("{}", key.public().to_ss58check());
+}
+
+fn list_accounts() {
+	let store = LocalKeystore::open(PathBuf::from(&KEYSTORE_PATH), None).unwrap();
+	println!("sr25519 keys:");
+	for pubkey in store.public_keys::<sr25519::AppPublic>().unwrap().into_iter() {
+		println!("{}", pubkey.to_ss58check());
+	}
+	println!("ed25519 keys:");
+	for pubkey in store.public_keys::<ed25519::AppPublic>().unwrap().into_iter() {
+		println!("{}", pubkey.to_ss58check());
+	}
+	drop(store);
+}
+
+fn print_metadata(cli: &Cli) {
+	let meta = get_chain_api(cli).get_metadata().unwrap();
+	println!("Metadata:\n {}", Metadata::pretty_format(&meta).unwrap());
+}
+
+fn print_sgx_metadata(cli: &Cli) {
+	let worker_api_direct = get_worker_api_direct(cli);
+	let metadata = worker_api_direct.get_state_metadata().unwrap();
+	println!("Metadata:\n {}", Metadata::pretty_format(&metadata).unwrap());
+}
+
+fn faucet(cli: &Cli, accounts: &[String]) {
+	let api = get_chain_api(cli);
+	let _api = api.set_signer(AccountKeyring::Alice.pair());
```

Comment:
Is `api` ever needed again in this function? If not, do `let api = api.set_signer()`. Otherwise, use a proper variable name like `api_with_alice_signer`.

The leading underscore notation in Rust denotes unused variables, so it is confusing twice over.